server/src/main/java/org/opensearch/cluster/routing/remote/InternalRemoteRoutingTableService.java

```diff
@@ -273,6 +326,63 @@ private void uploadIndex(
         }
     }
 
+    private void uploadIndexRoutingDiff(
```

Comment:
Lets take @Arpit-Bandejiya's changes from remote interface PR. This will significantly change the structure here and avoid code duplication

Reply:
Done.